### PR TITLE
feat(integration/s3): tempURL, cache, doc, integration

### PR DESCRIPTION
**Pull request by yesnault**
- [x] Temporary URL (to avoid send artifact to API, but direct upload to s3)
- [x] Cache
- [x] Doc on https://ovh.github.io/cds/docs/integrations/
- [x] Project Integration as Swift
- [x] Check json conf

Signed-off-by: Yvonnick Esnault <yvonnick@esnau.lt>

### CODE REVIEW

**Comment by yesnault** [comment]
new IT:
```
➜  cds git:(ye-reynn-master-s3) ✗ venom run --var cds.build.cdsctl=cdsctl tests/sc_workflow_storage_awss3_artifact.yml
SUCCESS tests/sc_workflow_storage_awss3_artifact.yml   22.21588221s

Total:6 Duration:22.218462937s
OK:6
KO:0
Skipped:0
TestSuite:1
TestCase:6
➜  cds git:(ye-reynn-master-s3) ✗ venom run --var cds.build.cdsctl=cdsctl tests/sc_workflow_storage_awss3_worker_artifact.yml 
SUCCESS tests/sc_workflow_storage_awss3_worker_artifact31.782532385s

Total:6 Duration:31.785587976s
OK:6
KO:0
Skipped:0
TestSuite:1
TestCase:6
➜  cds git:(ye-reynn-master-s3) ✗ venom run --var cds.build.cdsctl=cdsctl tests/sc_workflow_storage_awss3_worker_cache.yml 
SUCCESS tests/sc_workflow_storage_awss3_worker_cache.ym22.055815325s

Total:7 Duration:22.059108803s
OK:7
KO:0
Skipped:0
TestSuite:1
TestCase:7
```

File: docs/content/docs/integrations/aws/aws_s3.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+The AWS S3 Integration is a Self-Service integration that can be configured on a CDS Project.
+
+With this integration, you can use a dedicated AWS S3 Bucket Storage on :
+
+- action [Artifact Upload]({{< relref "/docs/actions/artifact-upload.md">}})
+- action [Artifact Downlaod]({{< relref "/docs/actions/artifact-download.md">}})
+- [worker cache command]({{< relref "/docs/components/worker/cache">}})
+
+Notice: per default, the storage is configured in CDS Configuration. This integration
```
Comment (bnjjj) [inline docs/content/docs/integrations/aws/aws_s3.md]: `by default` instead `per default` no ?

File: docs/content/docs/integrations/aws/aws_s3.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+card: 
+  name: storage
+---
+
+The AWS S3 Integration is a Self-Service integration that can be configured on a CDS Project.
+
+With this integration, you can use a dedicated AWS S3 Bucket Storage on :
+
+- action [Artifact Upload]({{< relref "/docs/actions/artifact-upload.md">}})
+- action [Artifact Downlaod]({{< relref "/docs/actions/artifact-download.md">}})
```
Comment (bnjjj) [inline docs/content/docs/integrations/aws/aws_s3.md]:
```suggestion
- action [Artifact Download]({{< relref "/docs/actions/artifact-download.md">}})
```

File: engine/api/objectstore/awss3.go [hunk 3]
```diff
@@ -109,8 +110,26 @@ func (s *AWSS3Store) Store(o Object, data io.ReadCloser) (string, error) {
...
+
+	if contentType != "" {
+		req.HTTPRequest.Header.Set("Content-Type", contentType)
+	}
+
+	urlStr, hs, err := req.PresignRequest(5 * time.Minute)
+	if err != nil {
+		return "", "", sdk.WrapError(err, "failed to sign request")
+	}
+	log.Debug("AWS-S3-Store> StoreURL urlStr:%v key:%v hs:%v", urlStr, *key, hs)
```
Comment (bnjjj) [inline engine/api/objectstore/awss3.go]: key could not be nil ?

**Pull request merged**